### ISSUE TITLE
Bug: sort-type-union-intersection-members breaks code combining & and |

### ISSUE THREAD

**Issue by ysulyma** [typescript-eslint/typescript-eslint#6158]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBAglC8UDeUCGAuKAiGWDcUAvgFCiRQBCCyUARplhfkaeNAMLUoDGD7zJVuQAi1ABScAPpQCUUAGSw8xIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZAe2jqNNtiaUOTWs3yJoyRGXxCRAW0TyARhNQZIE6F0jgwAXxD6gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type A = { a: "A"; }
type B = { b: "B"; }
type C = { c: "C"; }

// this line violates @typescript-eslint/sort-type-union-intersection-members
export type D = (C | B) & A;
```

### ESLint Config

```javascript
{
  "extends": [
    "plugin:@typescript-eslint/recommended"
  ],
  "rules": {
    "@typescript-eslint/sort-type-union-intersection-members": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected that it would change the line to
```ts
export type D = A & (B | C);
```

### Actual Result

It changed the line to
```ts
export type D = B | (A & C);
```
which has a different meaning from the original type.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.38.1` |
| `@typescript-eslint/parser`            | `5.38.1` |
| `TypeScript`                           | `4.8.3` |
| `ESLint`                               | `8.24.0` |
| `node`                                 | `16.17.0` |

**Comment by bradzacher** [comment]
This was fixed in https://github.com/typescript-eslint/typescript-eslint/pull/6118 and will be released with our next release.
